mico-core/src/main/java/io/github/ust/mico/core/broker/KafkaFaasConnectorDeploymentInfoBroker.java

```diff
@@ -83,4 +93,93 @@
+
+        eventuallyUpdateTopics(kfConnectorDeploymentInfoRequestDTO, storedServiceDeploymentInfo);
+        eventuallyUpdateOpenFaaSFunction(kfConnectorDeploymentInfoRequestDTO, storedServiceDeploymentInfo);
+        return deploymentInfoRepository.save(storedServiceDeploymentInfo);
+    }
+
+    private void eventuallyUpdateTopics(KFConnectorDeploymentInfoRequestDTO kfConnectorDeploymentInfoRequestDTO,
+                                        MicoServiceDeploymentInfo storedServiceDeploymentInfo) {
+        // update existing topics
+        List<MicoTopicRole> topics = storedServiceDeploymentInfo.getTopics();
+        for (MicoTopicRole role : topics) {
+            MicoTopic topic = role.getTopic();
+            switch (role.getRole()) {
+                case INPUT:
+                    topic.setName(kfConnectorDeploymentInfoRequestDTO.getInputTopicName());
+                    break;
+                case OUTPUT:
+                    topic.setName(kfConnectorDeploymentInfoRequestDTO.getOutputTopicName());
+                    break;
+                case DEAD_LETTER: //TODO
+                case INVALID_MESSAGE: //TODO
+                case TEST_MESSAGE_OUTPUT: //TODO
+            }
+        }
+        // add topics, if they do not exist, yet
+        List<MicoTopicRole.Role> storedTopicRoles = storedServiceDeploymentInfo.getTopics().stream().map(
+            MicoTopicRole::getRole).collect(Collectors.toList());
+
+        for (MicoTopicRole.Role role : MicoTopicRole.Role.values()) {
+            // TODO, currently the the KFConnectorDeploymentInfoRequestDTO only provides INPUT and OUTPUT.
+            //  As soon as it also provides DEAD_LETTER, INVALID_MESSAGE and TEST_MESSAGE_OUTPUT this should be
+            //  handled here as well
+
+            if (!storedTopicRoles.contains(role) && (role == INPUT || role == OUTPUT)) {
+                storedServiceDeploymentInfo.getTopics().add(
+                    new MicoTopicRole()
+                        .setRole(role)
+                        .setServiceDeploymentInfo(storedServiceDeploymentInfo)
+                        .setTopic(new MicoTopic().setName(
+                            (role == INPUT) ? kfConnectorDeploymentInfoRequestDTO.getInputTopicName()
+                                : kfConnectorDeploymentInfoRequestDTO.getOutputTopicName()
+                        )));
+            }
+        }
+    }
+
+    private void eventuallyUpdateOpenFaaSFunction(KFConnectorDeploymentInfoRequestDTO kfConnectorDeploymentInfoRequestDTO,
+                                                  MicoServiceDeploymentInfo storedServiceDeploymentInfo) {
+        OpenFaaSFunction openFaaSFunction = storedServiceDeploymentInfo.getOpenFaaSFunction();
+        if (openFaaSFunction == null) {
+            storedServiceDeploymentInfo.setOpenFaaSFunction(
+                new OpenFaaSFunction().setName(kfConnectorDeploymentInfoRequestDTO.getOpenFaaSFunctionName()));
```

Comment:
If a node with this OpenFaaSFunction already exists in the database, it should be reused.